src/components/CourseForm/CourseForm.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+import Input from '../../common/Input/Input';
+import Button from '../../common/Button/Button';
+import {
+	CREATE_COURSE,
+	CREATE_COURSE_TITLE,
+	UPDATE_COURSE,
+	ROUTE_COURSES,
+	ROUTE_LOGIN,
+} from '../../constants';
+import CreateAuthors from './components/CreateAuthors/CreateAuthors';
+import {
+	validateEmptyString,
+	validateEmptyList,
+	validateMinLength,
+} from '../../helpers/validationInputs';
+import store from '../../store';
+import getUsersSelector from '../../store/user/selectors';
+import {
+	createCourseThunk,
+	fillSelectedCourseThunk,
+} from '../../store/courses/thunk';
+import getCoursesSelector from '../../store/courses/selectors';
+import { setSelectedCourseProperty } from '../../store/courses/actions';
+import styles from './styles.module.scss';
+
+const CreateForm = () => {
+	const { courseId } = useParams();
+	const navigate = useNavigate();
+
+	const userData = useSelector(getUsersSelector);
+	const coursesData = useSelector(getCoursesSelector);
+	const { selectedCourse } = coursesData;
+
+	const validate = () => {
+		return (
+			validateEmptyString(selectedCourse.title) &&
+			validateEmptyString(selectedCourse.description) &&
+			validateMinLength(selectedCourse.description, 2) &&
+			validateEmptyList(selectedCourse.authors, 1) &&
+			Number(selectedCourse.duration) > 0
+		);
+	};
+
+	const createCourse = () => {
+		if (!validate()) {
+			alert('Please fill in all fields');
+			return;
+		}
+
+		store.dispatch(createCourseThunk({ ...selectedCourse }));
```

Comment:
new course is not shown in the courses page after redirect, either save it in store in dispatch or refetch it again